project_task_analytic_propagation/migrations/18.0.3.0.0/post-migration.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/LGPL-3.0)
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    env["account.analytic.plan"].search([])._sync_plan_column("project.task")
+    env.cr.execute("""
+        SELECT pt.id, pt.analytic_account_id_old, ac.plan_id
+            FROM project_task AS pt
+        INNER JOIN account_analytic_account as ac ON (ac.id=pt.analytic_account_id_old)
+            WHERE pt.analytic_account_id_old IS NOT NULL
+        """)
+    rows = env.cr.fetchall()
+    for row in rows:
+        task_id = row[0]
+        analytic_account = row[1]
+        plan_id = row[2]
```

Comment:
This is easier to unpack 😉 and way more readable
```suggestion
    for task_id, analytic_account, plan_id in rows:
```